src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -79,65 +88,82 @@ public virtual void FileWorkItems(string sarifLogFileContents)
 
         public virtual void FileWorkItems(SarifLog sarifLog)
         {
-            this.FilingSucceeded = false;
-            this.FiledWorkItems = new List<WorkItemModel>();
-
-            sarifLog = sarifLog ?? throw new ArgumentNullException(nameof(sarifLog));
-
-            this.FilingClient.Connect(this.FilingContext.PersonalAccessToken).Wait();
-
-            OptionallyEmittedData optionallyEmittedData = this.FilingContext.DataToRemove;
-            if (optionallyEmittedData != OptionallyEmittedData.None)
+            using (Logger.BeginScope(nameof(FileWorkItems)))
             {
-                var dataRemovingVisitor = new RemoveOptionalDataVisitor(optionallyEmittedData);
-                dataRemovingVisitor.Visit(sarifLog);
-            }
+                this.FilingSucceeded = false;
+                this.FiledWorkItems = new List<WorkItemModel>();
 
-            optionallyEmittedData = this.FilingContext.DataToInsert;
-            if (optionallyEmittedData != OptionallyEmittedData.None)
-            {
-                var dataInsertingVisitor = new InsertOptionalDataVisitor(optionallyEmittedData);
-                dataInsertingVisitor.Visit(sarifLog);
-            }
+                sarifLog = sarifLog ?? throw new ArgumentNullException(nameof(sarifLog));
 
-            SplittingStrategy splittingStrategy = this.FilingContext.SplittingStrategy;
-            if (splittingStrategy == SplittingStrategy.None)
-            {
-                FileWorkItemsHelper(sarifLog, this.FilingContext, this.FilingClient);
-                return;
-            }
-
-            IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+                Logger.LogInformation("Connecting to filing client: {accountOrOrganization}", this.FilingClient.AccountOrOrganization);
```

Comment:
>Logger.LogInformatio [](start = 16, length = 20)

This logging doesn't have an event id. Is that considered a poor practice?

Reply:
I think we're using the logging for two different purposes. One is for capturing specific events. e.g. the call to create the work item took 150 ms. The other purpose is for tracing. I think we need events ids for the former, but not for the latter. And I think this example is the latter, tracing. Perhaps this should have been a LogDebug or LogTrace instead. But I didn't want to set the default trace level to Debug or Trace.

---
In reply to: [396837768](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r396837768) [](ancestors = 396837768)

Reply:
ok! Yes, maybe rename or refactor these methods slightly. Later.

---
In reply to: [397177520](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r397177520) [](ancestors = 397177520,396837768)